### PR TITLE
feat: Bagel Image Understanding

### PR DESCRIPTION
**Pull request by pufanyi**
## Checklist

- [ ] Format your code
- [ ] Add unit tests
- [ ] Update documentation as needed, including docstrings or example tutorials.

## Summary by CodeRabbit

- New Features
  - Added configurable options for ViT layer selection and RoPE usage.
  - Introduced per-token CE loss reweighting for finer training control.
  - Enhanced image handling with dedicated VAE and ViT processing paths.

- Improvements
  - More informative error when image size is incompatible with patch size.
  - Adjusted default cfg interval for image generation to (0.0, 1.0).
  - Refined loss normalization and guards for clearer behavior and messages.

- Bug Fixes
  - Corrected typo in an error message.

### CODE REVIEW

File: src/lmms_engine/models/bagel/bagel.py [hunk 1]
```diff
@@ -357,26 +365,31 @@ def forward(
...
-                ce = ce.sum() * dist.get_world_size() / total_ce_tokens
+                # ce = ce.sum() * dist.get_world_size() / total_ce_tokens
+                ce = ce.sum() / total_ce_tokens
             loss_dict["ce"] = ce.detach()
             loss = loss + ce * self.config.ce_weight  # TODO: check if this is correct
         else:
-            assert not self.config.visual_und
+            assert (
+                not self.config.visual_und
+            ), "ce loss is not supported when visual_und is True"
```
Comment (kcz358) [inline src/lmms_engine/models/bagel/bagel.py:386]: Is this part wrong for the assertion? CE should be true in visual und right?
